src/cheri_insts.sail

```diff
@@ -438,35 +419,22 @@ union clause ast = CToPtr : (regidx, regidx, regidx)
  * register *rd* is set to 0, otherwise integer register *rd* is set to
  * *cs1*.**address** $-$ *cs2*.**base**.
  *
- * ## Exceptions
- *
- * An exception is raised if:
- *   - *cs2*.**tag** is not set.
- *   - *cs1*.**tag** is set and *cs1* is sealed.
- *
  * ## Notes
  *
- * - *cs2* being sealed will not cause an exception to be raised. This is for
- *   further study.
+ * - *cs2* being sealed will not set *rd* to 0. This is for further study.
  */
 function clause execute(CToPtr(rd, cs1, cs2)) = {
-  let (cs2_val, cs2_idx) = if unsigned(cs2) == 0 then (DDC, DDC_IDX)
-                                                 else (C(cs2), 0b0 @ cs2);
+  let cs2_val = if unsigned(cs2) == 0 then DDC else C(cs2);
   let cs1_val = C(cs1);
-  if not (cs2_val.tag) then {
-    handle_cheri_cap_exception(CapEx_TagViolation, cs2_idx);
-    RETIRE_FAIL
-  } else if cs1_val.tag & isCapSealed(cs1_val) then {
-    handle_cheri_reg_exception(CapEx_SealViolation, cs1);
-    RETIRE_FAIL
-  } else {
-    /* Note: returning zero for untagged values breaks magic constants such as SIG_IGN */
-    X(rd) = if not (cs1_val.tag) then
-              zeros()
-            else
-              cs1_val.address - getCapBaseBits(cs2_val);
-    RETIRE_SUCCESS
-  }
+
+  let inCap = clearTagIfSealed(cs1_val);
```

Comment:
Not needed

Reply:
Hmm, it does change the behaviour because of the inCap.tag case: if the authority was sealed do we want to give zeros or just give the pointer?

Reply:
That's true... the two should really go together. Morello takes the same approach and just doesn't care, 0 is based solely on the being-converted capability, not the authority, where only the base matters (if at all, since turning DDCBO off makes it a CGetAddr for the non-zero case).